packages/core/src/dag.js

```diff
@@ -0,0 +1,102 @@
+import * as API from '@ucanto/interface'
+import { create as createLink } from './link.js'
+import { sha256 } from 'multiformats/hashes/sha2'
+import * as MF from 'multiformats/interface'
+import * as CBOR from './cbor.js'
+
+/**
+ * @param {unknown} value
+ * @returns {IterableIterator<API.Block>}
+ */
+export const iterate = function* (value) {
+  if (
+    value &&
```

Comment:
hmm intention here was to pass anything and if it is an `IPLDView` it would iterate over it's blocks and if it isn't do nothing. It is to traverse arbitrary structures that may contain `IPLDView`s inside. We could instead have `isIPLDView(object)` predicate so it would be:

```ts
for (const [key, value] of Object.entries(something)) {
    if (isIPLDView(value)) {
        for (const block of value.iterateIPLDBlocks()) {
        }
    }
}
```

But I find it more convenient to do this instead

```ts
for (const [key, value] of Object.entries(something)) {
    for (const block of DAG.iterate(value)) {
    }
}
```

Reply:
Added a comment to make it bit more clear.